views/Main.react.js

```diff
@@ -574,8 +574,20 @@ class MainPage extends Component {
     default_acl = options;
     this.starAcl(options, this.state.star)
     this.setState({
-      totalStudies: acl.length,
-      totalSubjects: options.length,
+      // study determination: at least one upper case
+      totalStudies: acl.filter(s=> /[A-Z]/.test(s.study)).length,
+      // subject determination: at least 2 alpha and 3 numeric
+      totalSubjects: options.filter(s=> {
+        let alp=0
+        let num=0
+
+        for (let i=0; i<s.subject.length; i++) {
+          /[a-zA-Z]/.test(s.subject[i]) && alp++;
+          /[0-9]/.test(s.subject[i]) && num++;
+        }
+
+        return alp>=2 && num>=3 ? true : false
+      }).length,
```

Comment:
we can probably use `reduce` here. I'm also confused why this wouldn't be all subjects since the name is `totalSubjects` and is instead a filtered list of subjects.